content/pages/how-to/use-direct-upload-with-continuous-integration.md

```diff
@@ -0,0 +1,178 @@
+   steps:
+     - uses: actions/checkout@v3
+     - name: Publish
+       uses: cloudflare/wrangler-action@2.0.0
+       with:
+         apiToken: ${{ secrets.CF_API_TOKEN }}
+         accountId: ${{ secrets.CF_ACCOUNT_ID }}
+         command: pages publish --project-name=<project name>
+```
+ 
+In the above code block, you have set up an Action that runs when you push code to the repository; running on `ubuntu-latest`, it first checks out your repository with the latest version of Node.
+ 
+{{<Aside type="note" header="Note">}}
+ 
+If your project uses a different Node version less than 16, you will have to upgrade your Node version as Wrangler requires at least Node.js v16.7.0.
+ 
+{{</Aside>}}
+ 
+The next step is to install [Wrangler](/workers/cli-wrangler/) in your project, using the `cloudflare/wrangler-action@2.0.0` Action, which requires you to set up your environment variables.
+ 
+Finally, the Action makes sure you log in to your Account with your Cloudflare credentials stored as secrets in your project to publish your site with the `pages publish` command.
+ 
+
+## Using CircleCI for CI/CD
+ 
+[CircleCI](https://circleci.com/) is another continuous integration and continuous delivery (CI/CD) platform that allows you to automate your build, test, and deployment pipeline. It can be configured to efficiently run complex pipelines with caching, docker layer caching, resource classes, etc.
+ 
+Similar to GitHub Actions, CircleCI can use Wrangler to continuously deploy your projects each time to push to your code.
+ 
+### Add Cloudflare credentials to CircleCI
+ 
+After you have generated your Cloudflare API token and found your account ID in the dashboard, you will need to add them to your CircleCI dashboard to use your environment variables in your project.
+ 
+On the CircleCI web app, go to your project's settings. You can navigate to "Projects" on the side navigation and click the ellipsis button in the project's row. You will see the option to add Environment variables:
+ 
+![project setting env var](../media/project-settings-env-var-v2.png)
+ 
+Then click on **Environment Variables** in the side navigation > **Add variable** and enter the name and value of the new environment variable, which is your Cloudflare credentials (`CLOUDFLARE_ACCOUNT_ID` and `CLOUDFLARE_API_TOKEN`).  
+ 
+### Setup
+ 
+Create a `.circleci/config.yml` file at the root of your project. This file contains the jobs that will be executed based on the order of your workflow. In your `config.yml` file, copy the following content:
+ 
+```yaml
+---
+filename: .circleci/config.yml
+---
+version: 2.1
+jobs:
+ Publish-to-Pages:
+   docker:
+     - image: cimg/node:17.2.0
+   steps:
+     - checkout
+     # Run your project's build step
+     - run: npm install && npm run build
+     # Publish with wrangler
+     - run: npx wrangler pages publish dist --project-name=<PROJECT NAME> # Replace dist with the name of your build folder and input your project name 
```

Comment:
do we want to specify wrangler version here?

Reply:
No, I don't think we want to.